bundles/io/org.eclipse.smarthome.io.console/src/main/java/org/eclipse/smarthome/io/console/internal/extension/MetadataConsoleCommandExtension.java

```diff
@@ -71,7 +72,9 @@ public void execute(String[] args, Console console) {
                     if (args.length < 4) {
                         printUsage(console);
                     } else {
-                        addMetadata(console, args[1], args[2], args[3], args.length > 4 ? args[4] : null);
+                        addMetadata(console, args[1], args[2], args[3],
+                                args.length > 4 ? Arrays.stream(args, 4, args.length).collect(Collectors.joining())
+                                        : null);
```

Comment:
Just to get it right: The arguments 4-x are joined without any separator, correct?
So, if argument 4 is "1", argument 5 is "2" and argument 6 is "3", this will result into a string "123".

Reply:
yes, because it parses lines like these: 
`{ key1 = value1, key2      =value2, key3=value3}` 
I even trimm the arguments left and right of the `=`. 

It does not make sense to have arguments like `{ key    1 = 1 2 3 }` - whoich would end up in `{key1=123}`. 

Surely the parsing could be made more error proof but I don't think this is needed here.

Reply:
If that is the case we could also nearly drop this PR and live with the original implementation. Only the removal of the curly brackets would remain.